libs/wingsdk/src/target-tf-aws/function.ts

```diff
@@ -114,9 +115,11 @@ export class Function extends cloud.FunctionBase {
       role: this.role.name,
     });
 
+    const functionName = this.sanitizeName(`${this.node.id}-${this.node.addr}`);
+
     // Create Lambda function
     this.function = new LambdaFunction(this, "Default", {
-      functionName: this.sanitizeFunctionName(this.node.id),
+      functionName: functionName,
       s3Bucket: bucket.bucket,
```

Comment:
Can you allow setting the bucket name via a property too `deploymentBucketName`?